recognition/src/face_detection/face_detector_data_provider.cpp

```diff
@@ -76,24 +76,19 @@ void pcl::face_detection::FaceDetectorDataProvider<FeatureType, DataSet, LabelTy
     }
   }
 
-  for (size_t i = 0; i < files.size (); i++)
+  for (const auto &filename : files)
   {
-    std::stringstream filestream;
-    filestream << data_dir << "/" << files[i];
-    std::string file = filestream.str ();
+    std::string file = data_dir + "/" + filename;
```

Comment:
Isn't clang tidy going to complain about using strings for a single char?

Reply:
Not sure if there is a warning, but I didn't saw one until now. And btw: If I search for regex `"."`  in all `*.cpp;*.h;*.hpp;*.qrc;*.cu` files I have 2119 hit - so I believe there is no check.